[SPARK-28084][SQL] Resolving the partition column name based on the resolver in sql load command

sujith71955 opened the pull request:
### What changes were proposed in this pull request?

LOAD DATA command resolves the partition column name as case sensitive manner,
where as in insert commandthe partition column name will be resolved using 
the SQLConf resolver where the names will be resolved based on `spark.sql.caseSensitive` property. Same logic can be applied for resolving the partition column names in LOAD COMMAND.

### Why are the changes needed?

It's to handle the partition column name correctly according to the configuration.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Existing UT and manual testing.

sujith71955 committed:
[SPARK-28084][SPARK-SQL]Resolving the partition column name based on the resolver which follows spark.sql.caseSensitive property configuration value.
What changes were proposed in this pull request?
LOAD DATA command resolves the partition column name as case sensitive manner, where as in insert command
the partition column name will be resolved using the SQLConf resolver where the names will be  resolved  based on
spark.sql.caseSensitive property. same logic can be applied for resolving the partition column names in LOAD COMMAND.

How was this patch tested?
Existing UT and manual testing.

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/11870/
Test PASSed.

sujith71955 commented:
@dongjoon-hyun  Please review and let me know for any clarifications.

SparkQA commented:
**[Test build #106634 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/106634/testReport)** for PR 24903 at commit [`b9962eb`](https://github.com/apache/spark/commit/b9962eb13c9a25e6da2ae2358782838e7be82d2d).

SparkQA commented:
**[Test build #106634 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/106634/testReport)** for PR 24903 at commit [`b9962eb`](https://github.com/apache/spark/commit/b9962eb13c9a25e6da2ae2358782838e7be82d2d).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/106634/
Test FAILed.

dongjoon-hyun commented on sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala:
```diff
@@ -370,6 +372,8 @@ case class LoadDataCommand(
     Seq.empty[Row]
   }
+
+  private def resolver (conf: SQLConf) : Resolver = conf.resolver
```
Do we need this?

dongjoon-hyun commented on sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala:
```diff
@@ -296,8 +296,10 @@ case class LoadDataCommand(
       partition.get.keys.foreach { colName =>
-        if (!targetTable.partitionColumnNames.contains(colName)) {
+        if (!targetTable.partitionColumnNames
+          .exists(fieldName => resolverFunc(fieldName, colName))) {
```
indentation.

dongjoon-hyun reviewed (commented):
Could you add a test case for `LOAD DATA`?

sujith71955 commented:
@dongjoon-hyun  sure, will update the changes based on the comments

gengliangwang commented on sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala:
```diff
@@ -296,8 +296,10 @@ case class LoadDataCommand(
           s"do not match number of partitioned columns in table " +
           s"(${targetTable.partitionColumnNames.size})")
       }
+      val resolverFunc = resolver(sparkSession.sessionState.conf)
```
val resolver = spark.sessionState.conf.resolver

HyukjinKwon commented:
Can we add a simple test?

sujith71955 commented:
sorry folks, stuck up with some personal work, will update the PR soon. thanks for all the suggestions.

dongjoon-hyun commented:
Gentle ping, @sujith71955 .

sujith71955 committed:
[SPARK-28084][SPARK-SQL]Resolving the partition column name based on the resolver which follows spark.sql.caseSensitive property configuration value.
What changes were proposed in this pull request?
LOAD DATA command resolves the partition column name as case sensitive manner, where as in insert command
the partition column name will be resolved using the SQLConf resolver where the names will be  resolved  based on
spark.sql.caseSensitive property. same logic can be applied for resolving the partition column names in LOAD COMMAND.

How was this patch tested?
Existing UT and manual testing.

SparkQA commented:
**[Test build #107339 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107339/testReport)** for PR 24903 at commit [`cde7b2e`](https://github.com/apache/spark/commit/cde7b2e36f103409b034a5029d0ddf80ca88dd1e).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/12481/
Test PASSed.

sujith71955 commented:
@All : Handled the comments,  please review and let me know for any clarifications/suggestions. thanks

SparkQA commented:
**[Test build #107340 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107340/testReport)** for PR 24903 at commit [`b316b7b`](https://github.com/apache/spark/commit/b316b7ba883fbc69807bdc22f7b5decb77915fb5).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/12483/
Test PASSed.

SparkQA commented:
**[Test build #107340 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107340/testReport)** for PR 24903 at commit [`b316b7b`](https://github.com/apache/spark/commit/b316b7ba883fbc69807bdc22f7b5decb77915fb5).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

SparkQA commented:
**[Test build #107339 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107339/testReport)** for PR 24903 at commit [`cde7b2e`](https://github.com/apache/spark/commit/cde7b2e36f103409b034a5029d0ddf80ca88dd1e).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/107340/
Test PASSed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/107339/
Test PASSed.

sujith71955 commented:
@dongjoon-hyun  Gentle ping, Let me know for any inputs. thanks all for the valuable inputs.

dongjoon-hyun commented:
Retest this please.

SparkQA commented:
**[Test build #107718 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107718/testReport)** for PR 24903 at commit [`b316b7b`](https://github.com/apache/spark/commit/b316b7ba883fbc69807bdc22f7b5decb77915fb5).

dongjoon-hyun commented on sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala:
```diff
@@ -296,8 +296,10 @@ case class LoadDataCommand(
           s"do not match number of partitioned columns in table " +
           s"(${targetTable.partitionColumnNames.size})")
       }
+      val resolverFunc = sparkSession.sessionState.conf.resolver
```
```scala
- val resolverFunc = sparkSession.sessionState.conf.resolver
+ val resolver = sparkSession.sessionState.conf.resolver
```

sujith71955 replied:
fixed.

dongjoon-hyun commented on sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala:
```diff
@@ -296,8 +296,10 @@ case class LoadDataCommand(
       partition.get.keys.foreach { colName =>
-        if (!targetTable.partitionColumnNames.contains(colName)) {
+        if (!targetTable
+          .partitionColumnNames.exists(fieldName => resolverFunc(fieldName, colName))) {
```
```scala
- if (!targetTable.partitionColumnNames.exists(fieldName => resolverFunc(fieldName, colName))) {
-   .partitionColumnNames.exists(fieldName => resolverFunc(fieldName, colName))) {
+ if (!targetTable.partitionColumnNames.exists(resolver(_, colName))) {
```

sujith71955 replied:
fixed

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/12845/
Test PASSed.

dongjoon-hyun commented on sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala:
```diff
@@ -879,7 +879,11 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
       // scalastyle:off caselocale
-      orderedPartitionSpec.put(colName.toLowerCase, partition(colName))
+      partition.keys.foreach { partColName =>
+        if (partColName.toLowerCase == colName.toLowerCase()) {
```
Ur, according to the PR title, you wanted `Resolving the partition column name based on the resolver`.
But, this is completely ignore the `resolver` setting. Let's be clear what you need be consistent.

sujith71955 replied:
very valid question, the problem here is the partition spec  string wont match with the column name coming from the table metadata, so when we try to get the partition value detail from TablePartitionSpec based on table metadata column name  we get empty .this is why i am trying to get the partition value based on the column name specified in spec.

sujith71955 replied:
> I left a few comments, @sujith71955 . The UT is the most important comment. Could you update your PR, please?

I updated my PR based on above comments. thanks all for the valuable time and great suggestions.

maropu replied:
Is this part not related to the issue in a LOAD command? If so, I thinks it would be better to split this fix into two prs.

maropu replied:
I checked the code around and I think this is a bug caused by column name resolution in https://github.com/apache/spark/pull/24903/files#diff-159191585e10542f013cb3a714f26075L882. Could you make a pr for this issue only?

dongjoon-hyun commented on sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala:
```diff
@@ -290,6 +291,13 @@ class HiveCommandSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
+        checkAnswer(
+          sql("SELECT employeeID, employeeName FROM part_table WHERE C = '2' AND D = '1'"),
+          sql("SELECT * FROM non_part_table").collect())
+      }
```
This UT seems to succeed without your code. Please provide a valid UT to validate your contribution.

sujith71955 replied:
sorry it was a mistake :)  i missed load data command where we shall specify the partition column name with different case compare to the actual provided while creation. thanks for pointing out.

dongjoon-hyun reviewed (changes requested):
I left a few comments, @sujith71955 . The UT is the most important comment. Could you update your PR, please?

maropu commented on sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala:
```diff
@@ -29,7 +29,7 @@ import org.apache.hadoop.fs.{FileContext, FsConstants, Path}
 import org.apache.spark.sql.{AnalysisException, Row, SparkSession}
 import org.apache.spark.sql.catalyst.TableIdentifier
-import org.apache.spark.sql.catalyst.analysis.{NoSuchPartitionException, UnresolvedAttribute, UnresolvedRelation}
+import org.apache.spark.sql.catalyst.analysis.{NoSuchPartitionException, Resolver, UnresolvedAttribute, UnresolvedRelation}
```
nit: we don't need this import.

sujith71955 replied:
fixed

SparkQA commented:
**[Test build #107718 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107718/testReport)** for PR 24903 at commit [`b316b7b`](https://github.com/apache/spark/commit/b316b7ba883fbc69807bdc22f7b5decb77915fb5).
 * This patch **fails due to an unknown error code, -9**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/107718/
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/12863/
Test PASSed.

SparkQA commented:
**[Test build #107741 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107741/testReport)** for PR 24903 at commit [`b316b7b`](https://github.com/apache/spark/commit/b316b7ba883fbc69807bdc22f7b5decb77915fb5).

sujith71955 committed:
[SPARK-28084][SPARK-SQL]Resolving the partition column name based on the resolver which follows spark.sql.caseSensitive property configuration value.
What changes were proposed in this pull request?
LOAD DATA command resolves the partition column name as case sensitive manner, where as in insert command
the partition column name will be resolved using the SQLConf resolver where the names will be  resolved  based on
spark.sql.caseSensitive property. same logic can be applied for resolving the partition column names in LOAD COMMAND.

How was this patch tested?
Existing UT and manual testing.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/12868/
Test PASSed.

SparkQA commented:
**[Test build #107747 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107747/testReport)** for PR 24903 at commit [`386343a`](https://github.com/apache/spark/commit/386343a1b516c64d7e03208403c5725429d08c87).

SparkQA commented:
**[Test build #107741 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107741/testReport)** for PR 24903 at commit [`b316b7b`](https://github.com/apache/spark/commit/b316b7ba883fbc69807bdc22f7b5decb77915fb5).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/107741/
Test PASSed.

sujith71955 commented:
> Can we add a simple test?



done

dongjoon-hyun commented on sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala:
```diff
@@ -296,8 +296,10 @@ case class LoadDataCommand(
       partition.get.keys.foreach { colName =>
-        if (!targetTable.partitionColumnNames.contains(colName)) {
+        if (!targetTable
+          .partitionColumnNames.exists(fieldName => resolver(fieldName, colName))) {
```
Please see https://github.com/apache/spark/pull/24903#discussion_r303724838 . 
We need to change more~

sujith71955 replied:
done

SparkQA commented:
**[Test build #107747 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107747/testReport)** for PR 24903 at commit [`386343a`](https://github.com/apache/spark/commit/386343a1b516c64d7e03208403c5725429d08c87).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/107747/
Test PASSed.

sujith71955 committed:
[SPARK-28084][SPARK-SQL]Resolving the partition column name based on the resolver which follows spark.sql.caseSensitive property configuration value.
What changes were proposed in this pull request?
LOAD DATA command resolves the partition column name as case sensitive manner, where as in insert command
the partition column name will be resolved using the SQLConf resolver where the names will be  resolved  based on
spark.sql.caseSensitive property. same logic can be applied for resolving the partition column names in LOAD COMMAND.

How was this patch tested?
Existing UT and manual testing.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/12888/
Test PASSed.

SparkQA commented:
**[Test build #107769 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107769/testReport)** for PR 24903 at commit [`261009e`](https://github.com/apache/spark/commit/261009e0035f17d6c0c42431d6c747e39d7913e9).

SparkQA commented:
**[Test build #107769 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107769/testReport)** for PR 24903 at commit [`261009e`](https://github.com/apache/spark/commit/261009e0035f17d6c0c42431d6c747e39d7913e9).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/107769/
Test PASSed.

dongjoon-hyun commented:
@sujith71955 .
The same code path exists in both `loadPartition` and `loadDynamicPartitions`. Could you make a test case for both code branch? For `loadDynamicPartition`, you may need to use `INSERT INTO` statement.

sujith71955 commented:
> @sujith71955 .
> The same code path exists in both `loadPartition` and `loadDynamicPartitions`. Could you make a test case for both code branch? For `loadDynamicPartition`, you may need to use `INSERT INTO` statement.

INSERT INTO command is working fine, below flow shall take care in this scenario

![image](https://user-images.githubusercontent.com/12999161/61525928-06789600-aa37-11e9-9bed-dc0d51e9fcec.png)

sujith71955 committed:
[SPARK-28084][SPARK-SQL]Resolving the partition column name based on the resolver which follows spark.sql.caseSensitive property configuration value.
What changes were proposed in this pull request?
LOAD DATA command resolves the partition column name as case sensitive manner, where as in insert command
the partition column name will be resolved using the SQLConf resolver where the names will be  resolved  based on
spark.sql.caseSensitive property. same logic can be applied for resolving the partition column names in LOAD COMMAND.

How was this patch tested?
Existing UT and manual testing.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/13025/
Test PASSed.

SparkQA commented:
**[Test build #107913 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107913/testReport)** for PR 24903 at commit [`96c9537`](https://github.com/apache/spark/commit/96c9537590637b55c0f72f072c7581fbdf5d47c1).

SparkQA commented:
**[Test build #107913 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107913/testReport)** for PR 24903 at commit [`96c9537`](https://github.com/apache/spark/commit/96c9537590637b55c0f72f072c7581fbdf5d47c1).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/107913/
Test FAILed.

sujith71955 committed:
[SPARK-28084][SPARK-SQL]Resolving the partition column name based on the resolver which follows spark.sql.caseSensitive property configuration value.
What changes were proposed in this pull request?
LOAD DATA command resolves the partition column name as case sensitive manner, where as in insert command
the partition column name will be resolved using the SQLConf resolver where the names will be  resolved  based on
spark.sql.caseSensitive property. same logic can be applied for resolving the partition column names in LOAD COMMAND.

How was this patch tested?
Existing UT and manual testing.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/13029/
Test PASSed.

SparkQA commented:
**[Test build #107917 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107917/testReport)** for PR 24903 at commit [`a4be069`](https://github.com/apache/spark/commit/a4be069936f3411d45360c65c505c71f43676fc8).

dongjoon-hyun commented on sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala:
```diff
@@ -290,6 +291,29 @@ class HiveCommandSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
+    }
+  }
+
+  test("case insensitive property of partition column name in insert command") {
```
Since this is a bug fix, we need JIRA ID for the test case name.
```
- test("case insensitive property of partition column name in insert command") {
+ test("SPARK-28084 case insensitive property of partition column name in insert command") {
```

sujith71955 replied:
done

dongjoon-hyun commented on sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala:
```diff
@@ -290,6 +291,29 @@ class HiveCommandSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
+    withTable("part_table") {
+      withSQLConf(SQLConf.CASE_SENSITIVE.key -> "false") {
+        sql("CREATE TABLE part_table (price int, qty int) partitioned by (year int, month int)")
+        sql("INSERT INTO part_table PARTITION(YEar = 2015, month = 1) SELECT 1, 1")
```
BTW, this test case only covers `Static` Partition.
We need dynamic partitioning test case, too.

sujith71955 replied:
Added, please let me the know the added test-case is sufficient or not, thanks for your valuable suggestions and time.

sujith71955 committed:
[SPARK-28084][SPARK-SQL]Resolving the partition column name based on the resolver which follows spark.sql.caseSensitive property configuration value.
What changes were proposed in this pull request?
LOAD DATA command resolves the partition column name as case sensitive manner, where as in insert command
the partition column name will be resolved using the SQLConf resolver where the names will be  resolved  based on
spark.sql.caseSensitive property. same logic can be applied for resolving the partition column names in LOAD COMMAND.

How was this patch tested?
Existing UT and manual testing.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/13032/
Test PASSed.

SparkQA commented:
**[Test build #107921 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107921/testReport)** for PR 24903 at commit [`1fd9055`](https://github.com/apache/spark/commit/1fd90556bb780d6c5c59e5bb7523223269e6b1ea).

SparkQA commented:
**[Test build #107917 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107917/testReport)** for PR 24903 at commit [`a4be069`](https://github.com/apache/spark/commit/a4be069936f3411d45360c65c505c71f43676fc8).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/107917/
Test PASSed.

SparkQA commented:
**[Test build #107921 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107921/testReport)** for PR 24903 at commit [`1fd9055`](https://github.com/apache/spark/commit/1fd90556bb780d6c5c59e5bb7523223269e6b1ea).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/107921/
Test PASSed.

maropu commented on sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala:
```diff
@@ -290,6 +291,44 @@ class HiveCommandSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
+        }
+        checkAnswer(
+          sql("SELECT employeeID, employeeName FROM part_table WHERE c = '2' AND d = '1'"),
+          sql("SELECT * FROM non_part_table").collect())
```
nit: you don't need `collect()`.

maropu replied:
Is this LOAD issue specific to Hive? If not, could you move this test to `DDLParserSuite.scala`?

sujith71955 replied:
nit: you don't need collect(). Fixed

sujith71955 replied:
> Is this LOAD issue specific to Hive? If not, could you move this test to `DDLParserSuite.scala`?

Yes, i think Load commands are specific to Hive tables.

maropu replied:
oh, really? https://github.com/apache/spark/blob/a3bbc371cbdbcb469a181de8f58f83e193d4f341/sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLParserSuite.scala#L1505

sujith71955 replied:
you want me to move this testcase to DDLParserSuite.scala?

sujith71955 replied:
> oh, really?
> 
> https://github.com/apache/spark/blob/a3bbc371cbdbcb469a181de8f58f83e193d4f341/sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLParserSuite.scala#L1505

@maropu  sorry for misinterpettation, i moved the testcase to DDLParserSuite.scala, please let me know for any further clarifications. thanks for all valuable inputs.

sujith71955 replied:
@maropu @dongjoon-hyun 
There is a small problem, DDLParserSuite is a suite which basically does the verification of  parsing and plan resolution, no where i can find a testcase with execution flow, In my case since the resolution happens while Load command execution,  so i am not sure we shall use this suite for verifiction of this usecase, please suggest . Thanks

maropu commented on sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala:
```diff
@@ -290,6 +291,44 @@ class HiveCommandSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
+    // check for case insensitive property of partition column name in insert command.
+    withTable("part_table") {
+      val modeConfKey = "hive.exec.dynamic.partition.mode"
+      withSQLConf(SQLConf.CASE_SENSITIVE.key -> "false", modeConfKey -> "nonstrict") {
```
nit: 
```
      withSQLConf(
          SQLConf.CASE_SENSITIVE.key -> "false",
          "hive.exec.dynamic.partition.mode" -> "nonstrict") {
```

sujith71955 replied:
fixed

sujith71955 committed:
[SPARK-28084][SPARK-SQL]Resolving the partition column name based on the resolver which follows spark.sql.caseSensitive property configuration value.
What changes were proposed in this pull request?
LOAD DATA command resolves the partition column name as case sensitive manner, where as in insert command
the partition column name will be resolved using the SQLConf resolver where the names will be  resolved  based on
spark.sql.caseSensitive property. same logic can be applied for resolving the partition column names in LOAD COMMAND.

How was this patch tested?
Existing UT and manual testing.

sujith71955 commented on sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala:
```diff
@@ -290,6 +291,44 @@ class HiveCommandSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
+  }
+
+  test("SPARK-28084 case insensitive property of partition column name " +
+    "in insert command - dynamic partition") {
```
you are right, thanks for the suggestion :)

sujith71955 replied:
Handled the comments let me know for further suggestions/clarifications. thanks for the review.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/13076/
Test PASSed.

SparkQA commented:
**[Test build #107968 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107968/testReport)** for PR 24903 at commit [`c7c541e`](https://github.com/apache/spark/commit/c7c541e827353ae410c5ae4bd18ea097c43e83fd).

SparkQA commented:
**[Test build #107968 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/107968/testReport)** for PR 24903 at commit [`c7c541e`](https://github.com/apache/spark/commit/c7c541e827353ae410c5ae4bd18ea097c43e83fd).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/107968/
Test PASSed.

sujith71955 commented:
@dongjoon-hyun @maropu  Is PR looking good or anything i need to take care? please let me know.  Thanks

maropu commented on sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala:
```diff
@@ -289,7 +290,46 @@ class HiveCommandSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
+  test("SPARK-28084 case insensitive names of dynamic partitioning in INSERT commands") {
+    // check for case insensitive property of partition column name in insert command.
+    withTable("part_table") {
+      val modeConfKey = "hive.exec.dynamic.partition.mode"
```
nit: removes it.

maropu commented on sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala:
```diff
@@ -289,7 +290,46 @@ class HiveCommandSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
+        "hive.exec.dynamic.partition.mode" -> "nonstrict") {
+        sql("CREATE TABLE part_table (price int) partitioned by (year int)")
+        sql("INSERT INTO part_table PARTITION(YEar) SELECT 1, 2019")
+        checkAnswer(
```
nit: `        checkAnswer(sql("SELECT * FROM part_table"), Row(1, 2019))`

maropu commented on sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala:
```diff
@@ -289,7 +290,46 @@ class HiveCommandSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
+      withSQLConf(SQLConf.CASE_SENSITIVE.key -> "false") {
+        sql("CREATE TABLE part_table (price int, qty int) partitioned by (year int, month int)")
+        sql("INSERT INTO part_table PARTITION(YEar = 2015, month = 1) SELECT 1, 1")
+        checkAnswer(
```
nit: `        checkAnswer(sql("SELECT * FROM part_table"), Row(1, 1, 2015, 1))`

maropu commented on sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala:
```diff
@@ -289,7 +290,46 @@ class HiveCommandSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
+  }
+
+  test("SPARK-28084 case insensitive names of static partitioning in INSERT commands") {
+    // check for case insensitive property of partition column name in insert command.
```
nit: probably, we don't need this comment. (I think the test title already explains about it)

maropu commented on sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala:
```diff
@@ -289,7 +290,46 @@ class HiveCommandSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
+  }
+
+  test("SPARK-28084 case insensitive names of dynamic partitioning in INSERT commands") {
+    // check for case insensitive property of partition column name in insert command.
```
ditto

sujith71955 committed:
[SPARK-28084][SPARK-SQL]Resolving the partition column name based on the resolver which follows spark.sql.caseSensitive property configuration value.
What changes were proposed in this pull request?
LOAD DATA command resolves the partition column name as case sensitive manner, where as in insert command
the partition column name will be resolved using the SQLConf resolver where the names will be  resolved  based on
spark.sql.caseSensitive property. same logic can be applied for resolving the partition column names in LOAD COMMAND.

How was this patch tested?
Existing UT and manual testing.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/13185/
Test PASSed.

SparkQA commented:
**[Test build #108079 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108079/testReport)** for PR 24903 at commit [`6bb843c`](https://github.com/apache/spark/commit/6bb843c971a8fb161c9836141614593b9f8e277a).

sujith71955 commented:
@maropu  i updated the PR based on your comments. thanks :)

SparkQA commented:
**[Test build #108079 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108079/testReport)** for PR 24903 at commit [`6bb843c`](https://github.com/apache/spark/commit/6bb843c971a8fb161c9836141614593b9f8e277a).
 * This patch **fails due to an unknown error code, -9**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/108079/
Test FAILed.

sujith71955 commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/13206/
Test PASSed.

SparkQA commented:
**[Test build #108098 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108098/testReport)** for PR 24903 at commit [`6bb843c`](https://github.com/apache/spark/commit/6bb843c971a8fb161c9836141614593b9f8e277a).

SparkQA commented:
**[Test build #108098 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108098/testReport)** for PR 24903 at commit [`6bb843c`](https://github.com/apache/spark/commit/6bb843c971a8fb161c9836141614593b9f8e277a).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/108098/
Test PASSed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/13996/
Test PASSed.

SparkQA commented:
**[Test build #108922 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108922/testReport)** for PR 24903 at commit [`d18eeea`](https://github.com/apache/spark/commit/d18eeeab33e6780064f744fdc74a1ec33f81c75f).

SparkQA commented:
**[Test build #108922 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108922/testReport)** for PR 24903 at commit [`d18eeea`](https://github.com/apache/spark/commit/d18eeeab33e6780064f744fdc74a1ec33f81c75f).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/108922/
Test PASSed.

SparkQA commented:
**[Test build #108931 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108931/testReport)** for PR 24903 at commit [`e68cb88`](https://github.com/apache/spark/commit/e68cb886f66af53b79dfdecdabda396b93aaba67).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14005/
Test PASSed.

SparkQA commented:
**[Test build #108931 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/108931/testReport)** for PR 24903 at commit [`e68cb88`](https://github.com/apache/spark/commit/e68cb886f66af53b79dfdecdabda396b93aaba67).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/108931/
Test FAILed.

SparkQA commented:
**[Test build #109020 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109020/testReport)** for PR 24903 at commit [`04d9237`](https://github.com/apache/spark/commit/04d9237b2b28c181e438e9d67d67391b46d6d1e2).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14087/
Test PASSed.

SparkQA commented:
**[Test build #109022 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109022/testReport)** for PR 24903 at commit [`0284691`](https://github.com/apache/spark/commit/0284691d53faf93ceb10a4e1e731e1eb6106707d).

sujith71955 commented:
@dongjoon-hyun @maropu I updated the logic a bit, I am using PartitioningUtils.normalizePartitionSpec() for normalizing the paritionspec.  insert and all other flows uses same method for normalizing the paritionspec, let me know for any suggestions.

SparkQA commented:
**[Test build #109022 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109022/testReport)** for PR 24903 at commit [`0284691`](https://github.com/apache/spark/commit/0284691d53faf93ceb10a4e1e731e1eb6106707d).
 * This patch **fails due to an unknown error code, -9**.
 * This patch merges cleanly.
 * This patch adds no public classes.

SparkQA commented:
**[Test build #109020 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109020/testReport)** for PR 24903 at commit [`04d9237`](https://github.com/apache/spark/commit/04d9237b2b28c181e438e9d67d67391b46d6d1e2).
 * This patch **fails due to an unknown error code, -9**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109022/
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109020/
Test FAILed.

sujith71955 commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

SparkQA commented:
**[Test build #109032 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109032/testReport)** for PR 24903 at commit [`0284691`](https://github.com/apache/spark/commit/0284691d53faf93ceb10a4e1e731e1eb6106707d).

SparkQA commented:
**[Test build #109032 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109032/testReport)** for PR 24903 at commit [`0284691`](https://github.com/apache/spark/commit/0284691d53faf93ceb10a4e1e731e1eb6106707d).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109032/
Test PASSed.

sujith71955 commented:
gentle ping @dongjoon-hyun @maropu

sujith71955 committed:
[SPARK-28084][SPARK-SQL]Resolving the partition column name based on the resolver which follows spark.sql.caseSensitive property configuration value.
What changes were proposed in this pull request?
LOAD DATA command resolves the partition column name as case sensitive manner, where as in insert command
the partition column name will be resolved using the SQLConf resolver where the names will be  resolved  based on
spark.sql.caseSensitive property. same logic can be applied for resolving the partition column names in LOAD COMMAND.

How was this patch tested?
Existing UT and manual testing.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14239/
Test PASSed.

SparkQA commented:
**[Test build #109170 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109170/testReport)** for PR 24903 at commit [`de5f671`](https://github.com/apache/spark/commit/de5f671dcaead11837ad027614cb2aa75315abb5).

SparkQA commented:
**[Test build #109170 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109170/testReport)** for PR 24903 at commit [`de5f671`](https://github.com/apache/spark/commit/de5f671dcaead11837ad027614cb2aa75315abb5).
 * This patch **fails Scala style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109170/
Test FAILed.

SparkQA commented:
**[Test build #109176 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109176/testReport)** for PR 24903 at commit [`c503f5d`](https://github.com/apache/spark/commit/c503f5d419323a16442f313857b030d49f28995b).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14248/
Test PASSed.

SparkQA commented:
**[Test build #109176 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109176/testReport)** for PR 24903 at commit [`c503f5d`](https://github.com/apache/spark/commit/c503f5d419323a16442f313857b030d49f28995b).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109176/
Test FAILed.

sujith71955 committed:
[SPARK-28084][SPARK-SQL]Resolving the partition column name based on the resolver which follows spark.sql.caseSensitive property configuration value.
What changes were proposed in this pull request?
LOAD DATA command resolves the partition column name as case sensitive manner, where as in insert command
the partition column name will be resolved using the SQLConf resolver where the names will be  resolved  based on
spark.sql.caseSensitive property. same logic can be applied for resolving the partition column names in LOAD COMMAND.

How was this patch tested?
Existing UT and manual testing.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14331/
Test PASSed.

SparkQA commented:
**[Test build #109264 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109264/testReport)** for PR 24903 at commit [`9c96d61`](https://github.com/apache/spark/commit/9c96d619fdbb3c9bb11d499681d4adb7b5b084e4).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14332/
Test PASSed.

SparkQA commented:
**[Test build #109264 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109264/testReport)** for PR 24903 at commit [`9c96d61`](https://github.com/apache/spark/commit/9c96d619fdbb3c9bb11d499681d4adb7b5b084e4).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109264/
Test FAILed.

dilipbiswal commented:
retest this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14334/
Test PASSed.

SparkQA commented:
**[Test build #109266 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109266/testReport)** for PR 24903 at commit [`9c96d61`](https://github.com/apache/spark/commit/9c96d619fdbb3c9bb11d499681d4adb7b5b084e4).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14338/
Test PASSed.

SparkQA commented:
**[Test build #109270 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109270/testReport)** for PR 24903 at commit [`983d68b`](https://github.com/apache/spark/commit/983d68bb7471b11bd18ac23c399a295addab60d8).

SparkQA commented:
**[Test build #109266 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109266/testReport)** for PR 24903 at commit [`9c96d61`](https://github.com/apache/spark/commit/9c96d619fdbb3c9bb11d499681d4adb7b5b084e4).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109266/
Test FAILed.

SparkQA commented:
**[Test build #109270 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109270/testReport)** for PR 24903 at commit [`983d68b`](https://github.com/apache/spark/commit/983d68bb7471b11bd18ac23c399a295addab60d8).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109270/
Test FAILed.

SparkQA commented:
**[Test build #109567 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109567/testReport)** for PR 24903 at commit [`f11e815`](https://github.com/apache/spark/commit/f11e8157b78fe86ec24520673320807a9c337ffc).

SparkQA commented:
**[Test build #109567 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109567/testReport)** for PR 24903 at commit [`f11e815`](https://github.com/apache/spark/commit/f11e8157b78fe86ec24520673320807a9c337ffc).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109567/
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/14628/
Test PASSed.

SparkQA commented:
**[Test build #109571 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109571/testReport)** for PR 24903 at commit [`e2f962d`](https://github.com/apache/spark/commit/e2f962dd61b7d243afe3427e88988310f0e2740c).

SparkQA commented:
**[Test build #109571 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/109571/testReport)** for PR 24903 at commit [`e2f962d`](https://github.com/apache/spark/commit/e2f962dd61b7d243afe3427e88988310f0e2740c).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/109571/
Test FAILed.

sujith71955 commented:
@dongjoon-hyun  @maropu @HyukjinKwon @dilipbiswal  Updated the logic  a bit, currently i am normalizing the paritionspec using an existing API PartitioningUtils.normalizePartitionSpec().
Please let me know for any suggestions.

sujith71955 commented:
@maropu 
Moved the testcase to SQLQuerySuite.scala  as DDLParserSuite basically does the verification of parsing and plan resolution, nowhere i can find a testcase with execution flow, In my case the resolution happens while Load command execution.  Hope its fine, please suggest. Thanks

sujith71955 commented:
gentle ping @dongjoon-hyun @maropu @dilipbiswal @HyukjinKwon

sujith71955 commented:
gentle ping @dongjoon-hyun @maropu @dilipbiswal @HyukjinKwon

sujith71955 commented:
gentle ping @dongjoon-hyun @maropu @dilipbiswal @HyukjinKwon

sujith71955 commented:
gentle ping @dongjoon-hyun @maropu @dilipbiswal @HyukjinKwon

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/15679/
Test PASSed.

sujith71955 commented:
gentle ping @dongjoon-hyun @maropu

sujith71955 commented:
gentle ping @dongjoon-hyun @maropu @HyukjinKwon

HyukjinKwon commented:
retest this please

HyukjinKwon reviewed (approved):
Seems right to me. cc @viirya who wrote the codes too

SparkQA commented:
**[Test build #111566 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111566/testReport)** for PR 24903 at commit [`8c2a053`](https://github.com/apache/spark/commit/8c2a0531705a60be65a65138ca7a36390a76e875).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/16574/
Test PASSed.

SparkQA commented:
**[Test build #111566 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111566/testReport)** for PR 24903 at commit [`8c2a053`](https://github.com/apache/spark/commit/8c2a0531705a60be65a65138ca7a36390a76e875).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/111566/
Test PASSed.

viirya commented on sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala:
```diff
@@ -2012,6 +2012,26 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
+    withTempDir { dir =>
+      val path = dir.toURI.toString.stripSuffix("/")
+      val dirPath = dir.getAbsoluteFile
+      Files.append("1", new File(dirPath, s"part-r-000011"), StandardCharsets.UTF_8)
```
nit: can remove s.

sujith71955 replied:
updated. thanks

viirya commented on sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala:
```diff
@@ -2012,6 +2012,26 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
+              |PARTITIONED BY (d STRING)
+            """.stripMargin)
+          sql(s"LOAD DATA LOCAL INPATH '$path/part-r-000011' " +
+            s"INTO TABLE part_table PARTITION(D ='1')")
```
nit: can remove s.

sujith71955 replied:
updated. thanks

viirya replied:
only need to remove s at L2028, L2027 needs s"".

sujith71955 replied:
yeah, not observed that interpolation is happening in the statement (:, corrected it now 👍

viirya commented on sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala:
```diff
@@ -278,6 +278,13 @@ case class LoadDataCommand(
+      PartitioningUtils.normalizePartitionSpec(
+        spec,
+        targetTable.partitionColumnNames,
+        targetTable.identifier.quotedString,
```
We can use tableIdentwithDB.

sujith71955 replied:
updated. thanks

viirya reviewed (approved):
Thanks @HyukjinKwon pinging me. Not aware of this PR. There were good reviews from other reviewers. Just few comments about style. Otherwise looks good.

sujith71955 committed:
[SPARK-28084][SPARK-SQL]Resolving the partition column name based on the resolver which follows spark.sql.caseSensitive property configuration value.
What changes were proposed in this pull request?
LOAD DATA command resolves the partition column name as case sensitive manner, where as in insert command
the partition column name will be resolved using the SQLConf resolver where the names will be  resolved  based on
spark.sql.caseSensitive property. same logic can be applied for resolving the partition column names in LOAD COMMAND.

How was this patch tested?
Existing UT and manual testing.

SparkQA commented:
**[Test build #111653 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111653/testReport)** for PR 24903 at commit [`81e07f7`](https://github.com/apache/spark/commit/81e07f7ae82ce2a43b3c64d855a65ce261e08d0e).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/16649/
Test PASSed.

sujith71955 commented:
@viirya Fixed the comments. thanks for the suggestions.

SparkQA commented:
**[Test build #111654 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111654/testReport)** for PR 24903 at commit [`7eb3e3c`](https://github.com/apache/spark/commit/7eb3e3cb8444fcac219788cd2be8d2bae5881381).

AmplabJenkins commented:
Merged build finished. Test PASSed.

SparkQA commented:
**[Test build #111653 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111653/testReport)** for PR 24903 at commit [`81e07f7`](https://github.com/apache/spark/commit/81e07f7ae82ce2a43b3c64d855a65ce261e08d0e).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/111653/
Test FAILed.

SparkQA commented:
**[Test build #111654 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111654/testReport)** for PR 24903 at commit [`7eb3e3c`](https://github.com/apache/spark/commit/7eb3e3cb8444fcac219788cd2be8d2bae5881381).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/111654/
Test FAILed.

viirya commented on sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala:
```diff
@@ -2012,6 +2012,26 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
+              |CREATE TABLE part_table (c STRING)
+              |PARTITIONED BY (d STRING)
+            """.stripMargin)
+          sql("LOAD DATA LOCAL INPATH '$path/part-r-000011' " +
```
this s"" is needed.

sujith71955 replied:
yeah, not observed that interpolation is happening in the statement (: , corrected it now 👍

SparkQA commented:
**[Test build #111671 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111671/testReport)** for PR 24903 at commit [`ca4b85c`](https://github.com/apache/spark/commit/ca4b85c60b88e23ec82611b5915e73bc2edb760a).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/16666/
Test PASSed.

SparkQA commented:
**[Test build #111671 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111671/testReport)** for PR 24903 at commit [`ca4b85c`](https://github.com/apache/spark/commit/ca4b85c60b88e23ec82611b5915e73bc2edb760a).
 * This patch **fails due to an unknown error code, -9**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/111671/
Test FAILed.

viirya commented:
retest this please.

SparkQA commented:
**[Test build #111677 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111677/testReport)** for PR 24903 at commit [`ca4b85c`](https://github.com/apache/spark/commit/ca4b85c60b88e23ec82611b5915e73bc2edb760a).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/16671/
Test PASSed.

SparkQA commented:
**[Test build #111677 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111677/testReport)** for PR 24903 at commit [`ca4b85c`](https://github.com/apache/spark/commit/ca4b85c60b88e23ec82611b5915e73bc2edb760a).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/111677/
Test PASSed.

viirya commented:
Looks good to me now.

maropu commented:
The code itself looks good to me, too. Could you use the new PR template for better commit logs? https://github.com/apache/spark/blob/master/.github/PULL_REQUEST_TEMPLATE
cc: @HyukjinKwon @dongjoon-hyun

dongjoon-hyun reviewed (approved):
+1, LGTM.
Thank you so much, @sujith71955 , @maropu , @HyukjinKwon @viirya .
Sorry for being late. I updated the PR description, too.
Merged to master.

sujith71955 commented:
> +1, LGTM.
> Thank you so much, @sujith71955 , @maropu , @HyukjinKwon @viirya .
> Sorry for being late. I updated the PR description, too.
> Merged to master.

Thanks all for your valuable time and suggestions .
